main.go

```diff
@@ -135,51 +143,62 @@ func createBook(w http.ResponseWriter, r *http.Request) {
 	var newBook Book
 	err := json.NewDecoder(r.Body).Decode(&newBook) //Read the Json body and save the entry to newBook
 	if err != nil {
-		log.Println("error while decoding the json entry:", err)
-		w.WriteHeader(http.StatusBadRequest)
+		log.Println(err)
+		w.WriteHeader(http.StatusBadRequest) //SHOULDN'T IT BE STATUS INTERNAL SERVER ERROR?
 		return
 	}
 
-	blankFields := filledFields(newBook) //Verify if all entry fields are filled.
-	if blankFields != "" {
+	filled := filledFields(newBook) //Verify if all entry fields are filled.
+	if !filled {
+		error100, err := json.Marshal(errorResponse{100, "All the fields - name, price and inventory - must be filled correctly."})
```

Comment:
I see you using `json.Marshal` but we usually use `json.NewEncoder(w).Encode(errorResponse)`.
https://pkg.go.dev/encoding/json#Encoder

Reply:
Why?

Reply:
Less memory allocation and it makes the code cleaner, there is no need to create a bytes buffer just to write it once you can write directly to the response body with `json.NewEncoder`

Reply:
Ok